modules/learn/pages/clusters-and-availability/automatic-failover.adoc

```diff
@@ -8,39 +9,72 @@
+
+Below are the types of events that can trigger auto-failover processing on a node:
 
 * _Node failure_.
-A server-node within the cluster is unresponsive (due to a network failure, out-of-memory problem, or other node-specific issue).
+A server-node within the cluster is unresponsive (due to a network failure, very high CPU utilization problem, out-of-memory problem, or other node-specific issue). This means that the ns-server process on the node has not responded to the cluster manager for the user-specified configured amount of time -- the health of the services running on the node is unknown.
 
-* _Multiple node failure_.
+** Multiple node failure.
 Concurrent correlated failure of multiple nodes such as physical rack of machines or multiple virtual machines sharing a host.
 
-* _Disk read/write failure_.
-Attempts to read from or write to disk on a particular node have resulted in a significant rate of failure, for longer than a specified time-period.
-The node is removed by auto-failover, even though the node continues to be contactable.
+* _Data Service disk read/write issues_.
+Data Service disk read/write errors. Attempts by the Data Service to read from or write to disk on a particular node have resulted in a significant rate of failure (errors returned), for longer than a specified time-period.
+
+* _Index or Data Service running on the mode is non-responsive or unhealthy_.
+** Index Service non-responsiveness.  
+Index Service running on a node sends heartbeat messages to the cluster manager as an indication of its health.  If the Index Service fails to send a heartbeat, it is considered unhealthy, and if it stays unhealthy for the user-specified threshold time for auto-failover, the cluster manager will start the auto-failover checks for the node that the index service is on.
+** Data Service is unhealthy.  
+Besides the Data Service disk read/write issues configured monitoring for auto-failover, the Data Service running on a node can be deemed unhealthy per various other internal monitoring. If the Data Service stays unhealthy for the user-specified threshold time for auto-failover, the cluster manager will start the auto-failover checks for the node that the data service is on.
+
+Note that the Data Service and Index Service health for auto-failover uses the same Timeout value set for node unresponsiveness (see xref:learn:clusters-and-availability:automatic-failover.adoc[Configuring Auto-Failover] -- this is the user-specified threshold time for auto-failover mentioned in the Data Service and Index Service monitoring.
```

Comment:
Typo:  Missing closing parenthesis.
Note that the Data Service and Index Service health for auto-failover uses the same Timeout value set for node unresponsiveness (see xref:learn:clusters-and-availability:automatic-failover.adoc[Configuring Auto-Failover]**)**